### PR TITLE
[tools/dataset_converter] Add the mnist dataset

### PR DESCRIPTION
**Pull request by zetwhite**
This PR adds a mnist tensorflow dataset to the tf_dataset_converter.

ONE-DCO-1.0-Signed-off-by: SeungHui Youn <sseung.youn@samsung.com>

from draft : https://github.com/Samsung/ONE/pull/12694

### CODE REVIEW

File: tools/generate_datafile/tf_dataset_converter/datasets.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 class Mnist():
     def check(dataset_name):
-        supported_dataset_list = ['fashion_mnist']
+        supported_dataset_list = ['fashion_mnist', 'mnist']
```
Comment (zetwhite) [inline tools/generate_datafile/tf_dataset_converter/datasets.py:12]: * about TensorFlow datasets mnist dataset : https://www.tensorflow.org/datasets/catalog/mnist 

AFAIU, this tool helps preprocess(e.g normalize, one-hot encode) data and save it to the binary files. 
Like `fashion-mnist` dataset, the newly added `mnist` dataset needed the same preprocess.

So, Let's add `mnist` here to use mnist dataset on onert_train.

**Review by jyoungyun — APPROVED** [review]
LGTM

**Review by Aeren1564 — APPROVED** [review]
LGTM!

**Comment by glistening** [comment]
Title should be either "Add the mnist dataset" or "Add an mnist dataset". I think the former looks better.

Also, beyond this PR, the name of `generate_datafile` is too general. It would be good to include `train` because `tools` is not for only training.

**Comment by zetwhite** [comment]
> Title should be either "Add the mnist dataset" or "Add an mnist dataset". I think the former looks better.

Updated the title.

**Pull request merged**